lib/trino-cache/src/main/java/io/trino/cache/EvictableCache.java

```diff
@@ -142,12 +148,16 @@ public V get(K key)
         Token<K> newToken = new Token<>(key);
         Token<K> token = tokens.computeIfAbsent(key, _ -> newToken);
         try {
-            startLoading(token);
+            synchronized (token) {
+                startLoading(token);
+            }
```

Comment:
nt: the synchronization could be done inside startLoading.

but even then requires a commentary, it's not obvious why its needed
or maybe we jus replace ongoingLoads collection with a field inside Token

Reply:
good idea about removing `ongoingLoads`